### PR TITLE
watcher function to watch the resource using Kubernetes watch api.

### PR DESCRIPTION
**Pull request by myakove**
##### Short description:
Introduce watcher function to watch the resource using Kubernetes watch  api.

### CODE REVIEW

**Comment by myakove** [comment]
!verified

File: ocp_resources/resource.py [hunk 1]
```diff
@@ -822,6 +822,25 @@ def labels(self):
         """
         return self.instance["metadata"]["labels"]
 
+    def conditions_sampler(self, timeout):
```
Comment (rnetser) [inline ocp_resources/resource.py:825]: Gets the resource not specifically `conditions` - should either be renamed or change what is yielded

**Review by rnetser — CHANGES_REQUESTED** [review]
.

**Comment by myakove** [comment]
!verified

**Review by rnetser — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Review by rnetser — DISMISSED** [review]

**Comment by myakove** [comment]
!verified

**Review by rnetser — APPROVED** [review]

**Comment by myakove** [comment]
!verified

**Comment by myakove** [comment]
!cherry-pick v4.12

**Pull request merged**